src/wallet/wallet.cpp

```diff
@@ -2951,7 +2951,7 @@ std::shared_ptr<CWallet> CWallet::Create(WalletContext& context, const std::stri
         // ensure this wallet.dat can only be opened by clients supporting HD with chain split and expects no default key
         walletInstance->SetMinVersion(FEATURE_LATEST);
 
-        walletInstance->InitWalletFlags(wallet_creation_flags);
+        walletInstance->InitWalletFlags(wallet_creation_flags | WALLET_FLAG_GLOBAL_HD_KEY);
```

Comment:
This will set `WALLET_FLAG_GLOBAL_HD_KEY` for non-descriptor wallets as well, which is a minor bug in my opinion.

I think we should set the flag in `CreateWallet` function in the same place we force sqlite for descriptor wallets. This line begs the question why `wallet_creation_flags` doesn't have the `WALLET_FLAG_GLOBAL_HD_KEY` set already.

Reply:
Done as suggested

Reply:
This line is still here though.

Reply:
Oops, fixed.